release/apparmor.go

```diff
@@ -39,54 +41,141 @@ const (
 	UnknownAppArmor AppArmorLevelType = iota
 	// NoAppArmor indicates that apparmor is not enabled.
 	NoAppArmor
+	// UnusableAppArmor indicates that apparmor is enabled but cannot be used.
+	UnusableAppArmor
 	// PartialAppArmor indicates that apparmor is enabled but some
 	// features are missing.
 	PartialAppArmor
 	// FullAppArmor indicates that all features are supported.
 	FullAppArmor
 )
 
+func (l AppArmorLevelType) String() string {
+	switch l {
+	case UnknownAppArmor:
+		return "unknown"
+	case NoAppArmor:
+		return "none"
+	case UnusableAppArmor:
+		return "unusable"
+	case PartialAppArmor:
+		return "partial"
+	case FullAppArmor:
+		return "full"
+	}
+	return "???"
+}
+
 var (
-	appArmorLevel          AppArmorLevelType
-	appArmorSummary        string
+	// appArmorLevel contains the assessment of the "level" of apparmor support.
+	appArmorLevel AppArmorLevelType = UnknownAppArmor
+	// appArmorSummary contains a human readable description of the assessment.
+	appArmorSummary string
+	// appArmorKernelFeatures contains a list of kernel features that are supported.
+	// If the value is nil then the features were not probed yet.
+	appArmorKernelFeatures []string
+	// appArmorParserFeatures contains a list of parser features that are supported.
+	// If the value is nil then the features were not probed yet.
 	appArmorParserFeatures []string
 )
 
 // AppArmorLevel quantifies how well apparmor is supported on the
-// current kernel.
+// current kernel. The computation is performed once and
+// cached but it is somewhat cosly to perform.
 func AppArmorLevel() AppArmorLevelType {
 	if appArmorLevel == UnknownAppArmor {
-		appArmorLevel, appArmorSummary = probeAppArmor()
+		assessAppArmor()
 	}
 	return appArmorLevel
 }
 
 // AppArmorSummary describes how well apparmor is supported on the
-// current kernel.
+// current kernel. The computation is performed once and
+// cached but it is somewhat cosly to perform.
 func AppArmorSummary() string {
 	if appArmorLevel == UnknownAppArmor {
-		appArmorLevel, appArmorSummary = probeAppArmor()
+		assessAppArmor()
 	}
 	return appArmorSummary
 }
 
+// AppArmorFeatures returns a sorted list of apparmor features like
+// []string{"dbus", "network"}. The computation is performed once and
+// cached.
+func AppArmorKernelFeatures() []string {
+	if appArmorKernelFeatures == nil {
+		appArmorKernelFeatures = probeAppArmorKernelFeatures()
+	}
+	return appArmorKernelFeatures
+}
+
+// AppArmorParserFeatures returns a sorted list of apparmor parser features
+// like []string{"unsafe", ...}.  The computation is performed once and
+// cached but is is somewhat costly to perform.
+func AppArmorParserFeatures() []string {
+	if appArmorParserFeatures == nil {
+		appArmorParserFeatures = probeAppArmorParserFeatures()
+	}
+	return appArmorParserFeatures
+}
+
+// AppArmorFeatures is a deprecated name for AppArmorKernelFeatures.
+func AppArmorFeatures() []string {
```

Comment:
If it's deprecated, do we still need to have it around?

Reply:
The parts that remove it are in another branch. I wanted to limit this for just the release package.